js/src/controls/GeoportalControls.js

```diff
@@ -0,0 +1,189 @@
+            displayAltitude : true,
+            displayCoordinates : true,
+            editCoordinates : false,
+            crs : '',
+            label : '',
+            selectedtype: '',
+            altitude : {
+              triggerDelay : 100,
+              responseDelay : 500,
+              noDataValue : -99999,
+              noDataValueTolerance : 90000,
+              serviceOptions : {}
+           },
+          systems : [
+           {
+              crs : L.CRS.EPSG4326,
+              label : "Lon,Lat",
+              selectedtype : "Geographical"
+            },
+           {
+             crs : L.geoportalCRS.EPSG2154,
+              label : "Lambert 93",
+              selectedtype : "Metric"
+            }
+          ],
+          units : ["DEC", "DMS"]
+
+            };
+          }
+        }
+
+        export class LeafletGeoportalMousePositionView extends control.LeafletControlView {
+          initialize(parameters) {
+            super.initialize(parameters);
+            this.map_view = this.options.map_view;
+          }
+          create_obj() {
+            this.obj = L.geoportalControl.MousePosition({})
+          }
+        }
+
+        export class LeafletGeoportalElevationPathModel extends control.LeafletControlModel {
+          defaults() {
+            return {
+              ...super.defaults(),
+             _view_name: 'LeafletGeoportalElevationPathView',
+              _model_name: 'LeafletGeoportalElevationPathModel'
+              };
+            }
+          }
+
+          export class LeafletGeoportalElevationPathView extends control.LeafletControlView {
+            initialize(parameters) {
+              super.initialize(parameters);
+              this.map_view = this.options.map_view;
+            }
+            create_obj() {
+              this.obj = L.geoportalControl.ElevationPath({})
+            }
+          }
+
+          export class LeafletGeoportalIsocurveModel extends control.LeafletControlModel {
+            defaults() {
+              return {
+                ...super.defaults(),
+                _view_name: 'LeafletGeoportalIsocurve',
+                _model_name: 'LeafletGeoportalIsocurveModel',
+                collapsed : false,
+                methods : ["time", "distance"],
+                exclusions : {
+                     toll : true,
+                     bridge : false,
+                     tunnel : true
+                  },
+                graphs : ["Pieton", "Voiture"],
+                isocurveOptions : {},
+                autocompleteOptions : {}
+
+              }
+            }
+          }
+
+            export class LeafletGeoportalIsocurveView extends control.LeafletControlView {
+              initialize(parameters) {
+                super.initialize(parameters);
+                this.map_view = this.options.map_view;
+              }
+              create_obj() {
+                this.obj = L.geoportalControl.Isocurve(this.get_options())
```

Comment:
There is no option on the `GeoportalIsocurve` Python side?

Reply:
There are some options for Isocurve in geoportal.py (from line 130 to 137). As previously mentionned,  options are nevertheliss still missing for `LayerSwitcher` and `ElevationPath` controls. They will be implemented in a next commit.

Reply:
But they are not set as options (no `o=True` in `.tag()`).
Note that you can link to the code [like this](https://github.com/jupyter-widgets/ipyleaflet/blob/3c05e88a9fcae706b4c1136ce58780f48df37d78/ipyleaflet/geoportal.py#L130-L137), this makes it easier for reviewers.

Reply:
Yes you are right. Sorry I  forgot the `o=True`in the `.tag()`